### PR TITLE
Add FHIR latency metric

### PR DESCRIPTION
**Pull request by moria97**

### CODE REVIEW

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/FhirToDataLakeOrchestratorJob.cs [hunk 1]
```diff
@@ -153,12 +153,16 @@ await foreach (FhirToDataLakeProcessingJobInputData input in inputs.WithCancella
...
-                _result.CompleteTime = DateTimeOffset.UtcNow;
+                var processEndtime = DateTimeOffset.UtcNow;
+                _result.CompleteTime = processEndtime;
 
                 progress.Report(JsonConvert.SerializeObject(_result));
 
-                _diagnosticLogger.LogInformation("Finish FhirToDataLake job.");
-                _logger.LogInformation($"Finish FhirToDataLake orchestrator job {_jobInfo.Id}");
+                var jobLatency = processEndtime - _inputData.DataEndTime;
+                _metricsLogger.LogProcessLatencyMetric(jobLatency.TotalSeconds);
```
Comment (QuanWanxx) [inline FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/FhirToDataLakeOrchestratorJob.cs:162]: Does dicom orchestrator job need this?

Comment (moria97) [reply FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/FhirToDataLakeOrchestratorJob.cs:162]: Yes. You can add it when implementing the dicom metrics.

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/FhirToDataLakeOrchestratorJob.cs [hunk 2]
```diff
@@ -153,12 +153,16 @@ await foreach (FhirToDataLakeProcessingJobInputData input in inputs.WithCancella
...
+
+                _diagnosticLogger.LogInformation($"Finish FhirToDataLake job. Synced result data to {_inputData.DataEndTime}.");
```
Comment (QuanWanxx) [inline FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/FhirToDataLakeOrchestratorJob.cs:164]: Typo": "Synced result data to {_inputData.DataEndTime}"

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/FhirToDataLakeOrchestratorJob.cs (hunk 1 above)
Comment (hongyh13) [inline FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/FhirToDataLakeOrchestratorJob.cs:162]: seem the job latency is only meaningful for the incremental job?

**Review by hongyh13 — APPROVED** [review]

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/FhirToDataLakeOrchestratorJob.cs (hunk 1 above)
Comment (moria97) [reply FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/FhirToDataLakeOrchestratorJob.cs:162]: Yes. Currently it's coarse-grained to report latency at trigger level.

**Pull request merged**